OSOE-548: Upgrade to Orchard Core 1.6

Psichorex opened the pull request:
Fixes #378 

[OSOE-548](https://lombiq.atlassian.net/browse/OSOE-548)

[OSOE-548]: https://lombiq.atlassian.net/browse/OSOE-548?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Psichorex committed:
Updating Lombiq.OSOCE.Web to 1.6.

Psichorex committed:
Updating Lombiq.OSOCE.Samples to 1.6.

Psichorex committed:
Upgrading Lombiq.HelpfulLibraries to 1.6.

Psichorex committed:
Updating Lombiq.ChartJs to 1.6.

Psichorex committed:
Upgrading to 1.6. in HelpfulLibraries.

Psichorex committed:
Upgrading to 1.6 in DataTables

Psichorex committed:
Upgrading HE and ApplicationInsights to 1.6

Psichorex committed:
Upgrading Hosting Tenants to 1.6

Psichorex committed:
Upgrading Lombiq.JsonEditor to 1.6.

Psichorex committed:
Upgrading Lombiq.Privacy to 1.6.

Psichorex committed:
Upgrading Lombiq.TrainingDemo to OC 1.6

Psichorex committed:
Upgrading Lombiq.UIKit to OC 1.6

Psichorex committed:
Dev checkout in Lombiq.VueJs

Psichorex committed:
Upgrading Lombiq.VueJs to OC 1.6

Psichorex committed:
Upgrading Lombiq.BaseTheme to OC 1.6

Psichorex committed:
Upgrading Lombiq.Hosting.MediaTheme to OC 1.6

Psichorex committed:
Upgrading Lombiq.SetupExtensions to OC 1.6

Psichorex committed:
Dev checkout in Lombiq.UITestingToolbox

Psichorex committed:
Upgrading Lombiq.UITestingToolbox to OC 1.6

Psichorex committed:
Upgrading Lombiq.AuditTrailExtensions to OC 1.6

Psichorex committed:
Upgrading NodaTime

Psichorex committed:
Upgrading Lombiq.Hosting.BuildVersionDisplay to OC 1.6

Psichorex committed:
Upgrading ContentEditors and DataTables to OC 1.6

Psichorex committed:
Updating Lombiq.ContentEditors

Psichorex committed:
Adding some updates to VueJs and BuildVersionDisplay

Psichorex committed:
Updating UITT deprecated method.

Psichorex committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-548

Psichorex committed:
Consolidating packages.

Psichorex committed:
Fixing analyser warning in Lombiq.TrainingDemo

Psichorex committed:
Testing whether Twitter module produces broken behaviour.

Psichorex committed:
Trying test with different parameters.

Psichorex committed:
Using AddIndexProvider.

Psichorex committed:
Fixing UI test.

Psichorex committed:
Fixing warning.

Piedone commented on src/Lombiq.OSOCE.Web/Program.cs:
```diff
@@ -19,7 +19,7 @@
             new Dictionary<string, IEnumerable<string>>
             {
-                ["OrchardCore.Twitter"] = new[] { "Lombiq.UIKit", "Lombiq.ChartJs" },
+                ["OrchardCore.Twitter"] = new[] { "Lombiq.UIKit" },
```
You shouldn't change this. As I told under the issue: on `dev`, the UI test passes. If the UI test breaks with 1.6, then you should fix the underlying functionality, not employ a workaround like this.

Psichorex replied:
There is a problem related most probably to Orchard Core I am going to discuss now here in detail so you get a good understanding.

This is the code that the current dev contains:
![image](https://user-images.githubusercontent.com/92299130/235221237-3669d771-c968-4ac5-9a77-0c4bc55c9698.png)

Because the Conditinal Twitter module has 2 conditions the IFeatureEventHandler will run fully 2 times.
This is the first run :
![image](https://user-images.githubusercontent.com/92299130/235221694-0a51982a-1cde-4b41-94a3-d0b85ae04254.png)
Here you can see that the featuresToEnable contains Twitter only and it will call :
`await _shellFeatureManager.EnableFeaturesAsync(featuresToEnable)` 
OrchardCore.Twitter gets enabled here.

Problem roots here :
![image](https://user-images.githubusercontent.com/92299130/235222056-6dcacb97-2ef4-42e6-b435-e7646bfb0c62.png)
Here you see that the `shellDescriptor.Features` contains exactly 59 items.

After Twitter getting enabled and the IFeatureEventHandler runing the second time fully because it hit the second condition feature the shellDescriptor.Features will again contain 59 items. 

So basically it does not indicate that Twitter has been enabled.

So it tries to activate it again and this happens:
![image](https://user-images.githubusercontent.com/92299130/235222415-23876d4d-a27a-4e26-af4e-c9b896245b6e.png)

This was causing the UI Test problems but and this is an important but those test were flaky and they started passing after a certain time.
It was enough for me to take notice that something is fishy that's how I found out that this part is not functioning properly.

Psichorex replied:
Added a workaround that fixes the issue.

Piedone replied:
Thanks for taking the time to explain. However, three things to note:

1. Where do you see the test being flaky? Flaky means it sometimes fails, sometimes passes. I checked out the last 10 successful `dev` build (the most recent one was a timeout due to https://github.com/Lombiq/UI-Testing-Toolbox/issues/228) and none of them showed this test being flaky. I.e, it passed on the first time in each run. Or did it flakily pass after 1.6?
2. The goal of this issue is to upgrade the OSOCE solution to OC 1.6. If something used to work under 1.5 but then breaks 1.6 then that's something that needs to be fixed. Why it used to work, and whether the implementation was a good idea at that time might be interesting, but is ultimately beside the point; it needs to keep working. Finding incompatibilities and thus things breaking between OC versions is part and parcel of such an issue.
3. I don't know if you intentionally use the word "workaround" instead of something like "fix". If you're just using these interchangeably without a specific meaning, then no problem; otherwise, I'd like to clarify that in all such cases, we're seeking a permanent, proper fix to a given problem, not just a workaround (which is usually used to mean a temporary fix to get the system into a working order without actually fixing the underlying problem). It might be that for some reason it's infeasible to implement a good fix, and we need to be content with just a workaround, but this is not such a case.

Piedone commented on pnpm-lock.yaml:
```diff
@@ -1,4 +1,4 @@
-lockfileVersion: 5.4
+lockfileVersion: 5.3
```
Why the downgrade, and the other changes in this file necessary?

Psichorex replied:
It was probably unintentional as pnpm-lock.yaml is popping up constantly without me touching it.

0liver replied:
Update to pnpm@7, please.

Psichorex replied:
@0liver It won't actually let me update PNPM. 
Giving npm install -g pnpm@7 will produce some installing like behavior but even after that checking pnpm -v give me back pnpm 6.32.0.

Psichorex committed:
Reverting pnpm-lock

Psichorex committed:
Adding small fixes to BaseTheme and UITT.

Psichorex committed:
Reverting unnecessary csproj changes.

Psichorex committed:
Using AddIndexProvider in HL.

Psichorex committed:
Giving workaround for not being able to acquire enabled features from the shellDescriptor.

Psichorex committed:
Reverting program.cs to dev.

Psichorex committed:
Resolving merge conflicts.

Psichorex committed:
Fixing merge conflict.

Psichorex committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-548

Psichorex committed:
Resolving merge conflicts in submodules.

Psichorex committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-548

Psichorex committed:
Removing replacement RecipeHarvester in UITT.

Piedone commented on pnpm-lock.yaml:
```diff
@@ -1326,4 +1326,4 @@ packages:
   /yocto-queue/0.1.0:
     resolution: {integrity: sha512-rVksvsnNCdJ/ohGc6xgPwyN8eheCxsiLM8mxuE/t/mOVqJewPuO1miLpTHQiRgTKCLexL4MeAFVagts7HmNZ2Q==}
     engines: {node: '>=10'}
-    dev: true
```
Revert the whitespace change too.

Psichorex committed:
Optimizing loop in Lombiq.Hosting

Piedone committed:
Reverting PNPM lock whitespace change

Piedone committed:
Disabling Shape Tracing as long as it messes with titles

Psichorex committed:
Updating Lombiq.Hosting.

Psichorex committed:
Merging dev to submodules to resolve conflicts.

Psichorex committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-548

Psichorex committed:
Merge branch 'issue/OSOE-548' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-548

Psichorex committed:
Removing AddDataMigration workaround from submodules.

Psichorex committed:
Removing workaround from DataTables.

Psichorex committed:
Adding IngnoreSystemSettings = true to Localization settings.

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-548

# Conflicts:
#	src/Modules/Lombiq.TrainingDemo

Piedone committed:
Typo in Training Demo

[skip ci]

Psichorex committed:
Resolving merge conflict in submodules.

Psichorex committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-548

Psichorex committed:
Updating Lombiq.Hosting.Tenants

Psichorex committed:
Resolving merge conflict in DataTables.

Psichorex committed:
Updating FeaturesGuard in Hosting.Tenants.

Psichorex committed:
Resolving merge conflict in UITT.

Psichorex committed:
Reverting Program.cs.

Psichorex committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-548

Psichorex committed:
Updating FeaturesGuard handler.

Psichorex committed:
Removing analyser warning in Lombiq.Hosting.Tenants.

Psichorex committed:
Resolving merge conflict in Hosting.Tenants.

Psichorex committed:
Merge branch 'dev' into issue/OSOE-548

Psichorex committed:
Updating packages in Hosting.Tenants.

Psichorex committed:
Readding missing Async suffix.

Psichorex committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-548

Psichorex committed:
Updating Lombiq.Hosting.Tenants.

Psichorex committed:
Adding comment in Lombiq.Hosting.Tenants.

Piedone committed:
Docs in Lombiq.Hosting.Tenants

Psichorex committed:
Updating Lombiq.Hosting.Tenants.

Psichorex committed:
Merge branch 'issue/OSOE-548' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-548

Psichorex committed:
Updating FeaturesGuard.

Psichorex committed:
Updating Lombiq.Hosting.Tenants.

Psichorex committed:
Adding more checks in Lombiq.Lombiq.Hosting.Tenants.

Psichorex committed:
Merging dev into OSOE-548.

Psichorex committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-548

Piedone committed:
Fixing analyzer violation in Lombiq.Hosting.Tenants

Piedone committed:
Lombiq.Hosting.Tenants refactoring

Psichorex committed:
Updating Lombiq.Hosting.Tenants.

Psichorex committed:
Upgrading Lombiq.UITestingToolbox in csproj.

Psichorex committed:
Updating Lombiq.UITestingToolbox.

Psichorex committed:
Updating Lombiq.UITestingToolbox.

Psichorex committed:
Updating Lombiq.VueJs

Psichorex committed:
Resolving merge conflict.

Psichorex committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-548

Psichorex committed:
Consolidating Lombiq.Tests.UI packages.

Piedone reviewed (changes requested):
1. Be sure to update **all** Lombiq package references in all projects while doing a release gradually: https://lombiq.atlassian.net/wiki/spaces/DEV/pages/786857987/Managing+releases+of+open-source+projects#An-example-of-releasing-OSOCE-projects A lot of the Helpful Libraries package references our outdated. Search for `PackageReference Include="Lombiq` in the solution, and you shouldn't see any non-alpha version for those packages that you released.
2. Also update the NuGetTest solution to the latest packages. Without that, the new packages are not tested.

Piedone committed:
Lombiq.UITestingToolbox docs

[skip ci]

Psichorex committed:
Upgrading to latest Lombiq alpha releases in submodules.

Psichorex committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-548

Psichorex committed:
Merge branch 'issue/OSOE-548' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-548

Psichorex committed:
Upgrading packages in NugetSolution.

Psichorex committed:
Upgrading package in Lombiq.NodeJs.Extensions.

Psichorex committed:
Removing mistakenly included "v" from versions.

Psichorex committed:
Lombiq.Tests.UI

Psichorex committed:
Upgrading ChartJs to latest alpha release.

Psichorex committed:
Updating to latest Lombiq.Tests.UI alpha release.

Psichorex committed:
Resolving merge conflict.

Psichorex committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-548

Psichorex committed:
Upgrading packages in Lombiq.Hosting.Tenants.EnvironmentRobots.

Psichorex committed:
Testing nuget sln with windows runner only.

Psichorex committed:
Adding dotnet --list-sdks to action.yml.

Psichorex committed:
Adding print-config-summary.

Psichorex committed:
Resolving merge conflict.

Psichorex committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-548

Psichorex committed:
Adding print-config-summary to all builds.

Psichorex committed:
Resolving merge conflict.

Psichorex committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-548

Psichorex committed:
Resolving merge conflicts.

Psichorex committed:
Merge branch 'dev' into issue/OSOE-548

# Conflicts:
#	NuGetTest/src/Lombiq.OSOCE.NuGet.Web/Lombiq.OSOCE.NuGet.Web.csproj
#	NuGetTest/test/Lombiq.OSOCE.NuGet.Tests.UI/Lombiq.OSOCE.NuGet.Tests.UI.csproj

Psichorex committed:
Upgrading UITT and NugetPackage reference.

Psichorex committed:
Resolving merge conflict.

Psichorex committed:
Upgrading to latest Lombiq.Tests.UI alpha.

Psichorex committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-548

# Conflicts:
#	NuGetTest/src/Lombiq.OSOCE.NuGet.Web/Lombiq.OSOCE.NuGet.Web.csproj

Psichorex committed:
Upgrading to Latest Lombiq.Tests.

dministro committed:
Updating submodule:
 - Lombiq.HelpfulLibraries: Prerelease

dministro committed:
Upgrading Lombiq.HelpfulLibraries.* package reference version
Updating submodules:
 - Lombiq.HelpfulLibraries: prerelease
 - Lombiq.UITestingToolbox: Upgrading Lombiq.HelpfulLibraries.* Reverting Microsoft.Extensions.Configuration.* version changes

dministro committed:
Upgrading Lombiq.* package reference version
Updating submodules

dministro committed:
Updating submodule:
 - Lombiq.TrainingDemo: Reverting Microsoft.Extensions.* package reference version changes

dministro committed:
Upgrading Lombiq.ChartJs.* package reference version
Updating submodule:
 - Lombiq.ChartJs: Embedding debug info in all projects in NuGet build Disabling include symbol in publish NuGet workflow

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-548

dministro committed:
Merge branch 'issue/OSOE-548' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-641

dministro committed:
Updating submodule:
 - Lombiq.Hosting.Tenants: Fixing analyzer violation

dministro committed:
Updating submodule:
 - Lombiq.ChartJs: Adding comments

Piedone committed:
ChartJs docs

Piedone committed:
Updating submodules merged into issue/OSOE-548

[skip ci]

Piedone committed:
Merge pull request #492 from Lombiq/issue/OSOE-641

OSOE-641: Fix remaining issues for the Orchard Core 1.6 upgrade

Piedone reviewed (approved):
Just marking the merge from `issue/OSOE-641`.

Piedone commented on .github/workflows/build-and-test.yml:
```diff
@@ -20,6 +20,8 @@ jobs:
       ui-test-parallelism: 0
       build-create-binary-log: "true"
       blame-hang-timeout: "5m"
+      print-config-summary: "true"
```
Remove these before setting the issue to Done.

Psichorex committed:
Fixing FeatureGuard's local boolean.

Psichorex committed:
Removing print-config-summary configurations.

Psichorex committed:
Resolving merge conflict.

Psichorex committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-548

Psichorex committed:
Removing empty lines.

Piedone committed:
Revert "Removing print-config-summary configurations."

This reverts commit 6138b60298956cb8be007bfc8e9c4e403eca02ed.

Piedone committed:
More GHA issue branch references

Piedone committed:
Reverting issue branch references, removing print-config-summary

Piedone committed:
Releasing Lombiq.Npm.Targets and Lombiq.NodeJs.Extensions alphas

Psichorex committed:
Updating Lombiq.Hosting.Tenants.

Piedone committed:
Renaming variable Lombiq.Hosting.Tenants

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-548

# Conflicts:
#	src/Libraries/Lombiq.HelpfulLibraries
#	src/Libraries/Lombiq.OrchardCoreApiClient
#	test/Lombiq.UITestingToolbox
#	tools/Lombiq.GitHub.Actions

Piedone committed:
Merge branch 'dev' into issue/OSOE-548

# Conflicts:
#	src/Modules/Lombiq.Hosting.Tenants
#	test/Lombiq.UITestingToolbox

Piedone committed:
OrchardCoreApiClient method obsoletion

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-548

# Conflicts:
#	src/Libraries/Lombiq.HelpfulLibraries
#	src/Themes/Lombiq.BaseTheme
#	test/Lombiq.UITestingToolbox

Piedone committed:
Temporarily disabling .NET consolidation

Piedone committed:
Syntax

Piedone committed:
Re-enabling consolidation

Piedone committed:
Updating Lombiq.HelpfulLibraries NuGet reference to latest alpha

Piedone committed:
Fixing NuGetBuild property

sarahelsaig committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-548

# Conflicts:
#	src/Modules/Lombiq.Hosting.Tenants

sarahelsaig committed:
Fix menu address pointing to admin.

sarahelsaig committed:
Add UI test.

sarahelsaig committed:
Code styling.

sarahelsaig committed:
use string localizer

sarahelsaig committed:
Use SetContentPickerByDisplayTextAsync extension

sarahelsaig committed:
Respect admin prefix.

sarahelsaig committed:
Add GetCookieContainer extension.

sarahelsaig committed:
Extract base theme dependency verification.

sarahelsaig committed:
Add FetchWithBrowserContextAsync extension method.

sarahelsaig committed:
Add missing Async suffix.

sarahelsaig committed:
Actually disable cert checking.

sarahelsaig committed:
Code cleanup.

sarahelsaig committed:
Add missing async suffix.

sarahelsaig committed:
Add comment.

Piedone committed:
Updating submodules to merged dev

[skip ci]

Piedone committed:
Merge pull request #503 from Lombiq/issue/OSOE-647

OSOE-647: ⭐ Under Orchard Core 1.6, menu items generated via MainMenuNavigationProvider are prefixed with /Admin in Lombiq.BaseTheme

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-548

# Conflicts:
#	NuGetTest/src/Lombiq.OSOCE.NuGet.Web/Lombiq.OSOCE.NuGet.Web.csproj
#	src/Modules/Lombiq.ChartJs
#	src/Modules/Lombiq.ContentEditors
#	src/Modules/Lombiq.DataTables
#	src/Modules/Lombiq.HelpfulExtensions
#	src/Modules/Lombiq.Hosting.Azure.ApplicationInsights
#	src/Modules/Lombiq.Hosting.BuildVersionDisplay
#	src/Modules/Lombiq.Hosting.Tenants
#	src/Modules/Lombiq.Privacy
#	src/Modules/Lombiq.UIKit
#	src/Modules/Lombiq.VueJs
#	src/Themes/Lombiq.BaseTheme
#	src/Themes/Lombiq.Hosting.MediaTheme
#	test/Lombiq.UITestingToolbox

Piedone committed:
Cleaning up OC package versions

Piedone committed:
Updating Lombiq NuGet references

Piedone committed:
Lombiq.UIKit docs

Piedone committed:
Releasing NuGet alphas for all projects

Piedone committed:
Updating Lombiq NuGet references to release versions
